File: reverseStuff.py
def oneFunctionToRuleThemAll(stuff):
    if len(stuff.strip().split(" ")) < 2:
        stringList = list(stuff.strip())
        for i in range(int(len(stringList)/2)):
            temp = stringList[i]
            stringList[i] = stringList[len(stringList) - 1 - i]
            stringList[len(stringList) - 1 - i] = temp
        # print(stringList)    
        return "".join(stringList)
        
    stringList = stuff.strip().split(" ")
    print(stringList)
    for i in range(int(len(stringList)/2)):
        temp = stringList[i]
        stringList[i] = stringList[int(len(stringList)/2) - 1 - i]
        stringList[int(len(stringList)/2) - 1 - i] = temp
        
        
    return " ".join(stringList)

string = "hello there"
print(f"oneFunctionToRuleThemAll input: {string}\noneFunctionToRuleThemAll output: {oneFunctionToRuleThemAll(string)}\n")

biggerStringStuff = "regular sentnce pls"
print(f"oneFunctionToRuleThemAll input: {biggerStringStuff}:\noneFunctionToRuleThemAll output: {oneFunctionToRuleThemAll(biggerStringStuff)}\n")

string = "      hello       "
print(f"oneFunctionToRuleThemAll input: {string}\noneFunctionToRuleThemAll output: {oneFunctionToRuleThemAll(string)}\n")

biggerStringStuff = "    I    don't    no   hao  2 spel    "
print(f"oneFunctionToRuleThemAll input: {biggerStringStuff}:\noneFunctionToRuleThemAll output: {oneFunctionToRuleThemAll(biggerStringStuff)}\n")

File: reverseStuff.js
function oneFunctionToRuleThemAll(stuff) {
    if (stuff.trim().split(" ").length < 2) {
        let stringList = stuff.trim().split("");
        for (let i = 0; i < stringList.length/2; ++ i) {
            let temp = stringList[i];
            stringList[i] = stringList[stringList.length - 1 - i];
            stringList[stringList.length - 1 - i] = temp;
        }
        return stringList.join("");
    }
    let splitStringArray = stuff.trim().split(" ");
    for (let i = 0; i < splitStringArray.length/2; ++ i) {
        let temp = splitStringArray[i];
        splitStringArray[i] = splitStringArray[splitStringArray.length - 1 - i];
        splitStringArray[splitStringArray.length - 1 - i] = temp;
    }
    return splitStringArray.join(" ");  
}
let string = "hello there";
console.log(`\noneFunctionToRuleThemAll input: ${string}\noneFunctionToRuleThemAll output: ${oneFunctionToRuleThemAll(string)}\n`);

let biggerStringStuff = "regular sentnce pls";
console.log(`\noneFunctionToRuleThemAll input: ${biggerStringStuff}:\noneFunctionToRuleThemAll output: ${oneFunctionToRuleThemAll(biggerStringStuff)}\n`);

string = "      hello       ";
console.log(`\noneFunctionToRuleThemAll input: ${string}\noneFunctionToRuleThemAll output: ${oneFunctionToRuleThemAll(string)}\n`);

biggerStringStuff = "    I    don't    no   hao  2 spel    ";
console.log(`\noneFunctionToRuleThemAll input: ${biggerStringStuff}:\noneFunctionToRuleThemAll output: ${oneFunctionToRuleThemAll(biggerStringStuff)}\n`);
